tasks.py

```diff
@@ -724,20 +724,26 @@ def ruff(context, action=None, target=None, fix=False, output_format="concise"):
     if not target:
         target = ["."]
 
+    exit_code = 0
+
     if "format" in action:
         command = "ruff format "
         if not fix:
             command += "--check "
         command += " ".join(target)
-        run_command(context, command, warn=True)
+        if not run_command(context, command, warn=True):
+            exit_code = 1
 
     if "lint" in action:
         command = "ruff check "
         if fix:
             command += "--fix "
         command += f"--output-format {output_format} "
         command += " ".join(target)
-        run_command(context, command, warn=True)
+        if not run_command(context, command, warn=True):
+            exit_code = 1
+
+    raise Exit(code=exit_code)
```

Comment:
Is this right to raise the Exit with a successful exit code?

Reply:
Worth stealing code from core here?

```py
    if not (format_result and lint_result):
        if not fix:
            raise Exit("'ruff format' and/or 'ruff check' failed; you may want to run 'invoke ruff --fix'", code=1)
        raise Exit("'ruff format` and/or 'ruff check' failed; please see above for specifics", code=1)
    print("ruff successful!")
```

Reply:
> Is this right to raise the Exit with a successful exit code?

This is standard for pyinvoke

Reply:
> Worth stealing code from core here?
> 
> ```python
>     if not (format_result and lint_result):
>         if not fix:
>             raise Exit("'ruff format' and/or 'ruff check' failed; you may want to run 'invoke ruff --fix'", code=1)
>         raise Exit("'ruff format` and/or 'ruff check' failed; please see above for specifics", code=1)
>     print("ruff successful!")
> ```

Have we encountered a silent failure in ruff that required adding these messages? I've always seen ruff print a message to stdout/stderr when something fails.